CodeStarterCamp_Week4/main.swift

```diff
@@ -80,52 +86,174 @@ class BodyCondition {
+    var returnOrdinal = ""
+    
+    if num >= 10 && num < 100 {
+        switch num / 10 {
+        case 1:
+            returnOrdinal = "열"
+        case 2:
+            returnOrdinal = "스물"
+        case 3:
+            returnOrdinal = "서른"
+        case 4:
+            returnOrdinal = "마흔"
+        case 5:
+            returnOrdinal = "쉰"
+        case 6:
+            returnOrdinal = "예순"
+        case 7:
+            returnOrdinal = "일흔"
+        case 8:
+            returnOrdinal = "여든"
+        case 9:
+            returnOrdinal = "아흔"
+        default:
+            returnOrdinal = ""
+        }
+    }
+    
+    switch num % 10 {
+    case 1:
+        if num < 10 {
+             returnOrdinal += "첫 "
+        } else {
+            returnOrdinal += "한 "
+        }
+    case 2:
+        returnOrdinal += "두 "
+    case 3:
+        returnOrdinal += "세 "
+    case 4:
+        returnOrdinal += "네 "
+    case 5:
+        returnOrdinal += "다섯 "
+    case 6:
+        returnOrdinal += "여섯 "
+    case 7:
+        returnOrdinal += "일곱 "
+    case 8:
+        returnOrdinal += "여덟 "
+    case 9:
+        returnOrdinal += "아홉 "
+    default:
+        returnOrdinal += ""
+    }
+        
+    return returnOrdinal
+}
+
+func startRoutines (countRoutines: Int, routine: Routine, mybodyCondition: BodyCondition) {
+    do {
+        for countRoutine in 1...countRoutines {
+            print("\(changeNumToOrdinal(num: countRoutine))번째 \(routine.name)을(를) 시작합니다.")
+            for routineActivity in routine.activities {
+                try routineActivity.action(myBodyCondition)
+            }
+        }
+    } catch {
+        print("피로도가 100 이상입니다. 루틴을 중단합니다.")
+    }
+}
+
+func takeInputCount () -> Int {
+    var countRoutines = 0
+    print("루틴을 몇 번 반복할까요?")
+
+    do {
+        repeat {
+            do {
+                let inputCountRoutines = readLine()
+                countRoutines = try checkInputCount(inputCountRoutines: inputCountRoutines)
+            } catch {
+                print("잘못된 입력 형식입니다. 다시 입력해주세요.")
+            }
+        } while countRoutines == 0
+    }
+    
+    return countRoutines
+}
+
+struct Routine {
+    let name: String
+    let activities: [Activity]
+    
+    init(name: String, activities: [Activity]) {
+        self.name = name
+        self.activities = activities
+    }
+}
+
 struct Activity {
     let name: String
-    let action: (BodyCondition) -> Void
+    let action: (BodyCondition) throws -> Void
 }
 
-let 윗몸일으키기: Activity = Activity(name: "윗몸일으키기", action: { bodyCondition in
-    print("<<\(윗몸일으키기.name)을(를) 시작합니다>>")
-    bodyCondition.상체근력 += Int.random(in: 10...20)
-    bodyCondition.피로도 += Int.random(in: 10...20)
+let 윗몸일으키기: Activity = Activity(name: "윗몸일으키기", action: { bodyCondition throws in
+    doActivity(bodyCondition: bodyCondition, activityName: 스쿼트.name, 상체근력: 10...20, 하체근력: 0...0, 근지구력: 0...0, 피로도: 10...20)
+    guard bodyCondition.피로도 < 100 else {
+        throw ActivityError.overFatigue
+    }
```

Comment:
```suggestion
let 윗몸일으키기: Activity = Activity(
    name: "윗몸일으키기", 
    action: { bodyCondition throws in
        doActivity(
            bodyCondition: bodyCondition, 
            activityName: 스쿼트.name, 
            상체근력: 10...20, 
            하체근력: 0...0, 
            근지구력: 0...0, 
            피로도: 10...20
        )
        guard bodyCondition.피로도 < 100 else {
            throw ActivityError.overFatigue
        }
```

이런식으로도 줄바꿈을 활용할 수 있어요 ☺️